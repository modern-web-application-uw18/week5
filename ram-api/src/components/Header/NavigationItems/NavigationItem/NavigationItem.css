.NavigationItem {
    margin: 0;
    display: flex;
    width: auto;
    align-items: center;
    text-decoration: none;
}

.NavigationItem a {
    color: white;
    padding: 16px 10px;
    border-bottom: 4px solid transparent;
    text-decoration: none;
}

.NavigationItem a:hover,
.NavigationItem a:active,
.NavigationItem a.active {
    background-color: #272727;
    border-bottom: 4px solid rgb(206, 24, 109);
    color: white;
}