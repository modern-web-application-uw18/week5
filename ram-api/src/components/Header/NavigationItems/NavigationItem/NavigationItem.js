import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './NavigationItem.css';

const navigationItem = ( props ) => (
    <li className="NavigationItem">
        <NavLink to={props.link} exact={props.exact}>{props.children}</NavLink>
    </li>
);

navigationItem.propTypes = {
    link: PropTypes.string.isRequired
}

export default navigationItem;