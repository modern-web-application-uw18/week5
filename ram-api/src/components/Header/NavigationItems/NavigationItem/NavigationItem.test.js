import React from 'react';
import { mount } from 'enzyme';
import NavigationItem from './NavigationItem'
import { MemoryRouter} from 'react-router-dom';
// import characters from '../../__mocks__/characters.json'

describe('<NavigationItem />', () => {
    it('Should render NavigationItem', () => {

        const wrapper = mount(
            <MemoryRouter> 
                <NavigationItem link="/detail" />
            </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




