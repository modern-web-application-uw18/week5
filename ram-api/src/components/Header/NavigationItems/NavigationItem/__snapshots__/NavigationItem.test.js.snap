// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> Should render NavigationItem 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "nf7yv9",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "nf7yv9",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <navigationItem
      link="/detail"
    >
      <li
        className="NavigationItem"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/detail"
        >
          <Route
            path="/detail"
          >
            <Link
              aria-current={false}
              replace={false}
              to="/detail"
            >
              <a
                aria-current={false}
                href="/detail"
                onClick={[Function]}
              />
            </Link>
          </Route>
        </NavLink>
      </li>
    </navigationItem>
  </Router>
</MemoryRouter>
`;
